src/test/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotPopAndLockDatasetStepTest.java

```diff
@@ -47,44 +51,74 @@ class DeleteSnapshotPopAndLockDatasetStepTest {
   @BeforeEach
   void setUp() {
     snapshotId = UUID.randomUUID();
+    datasetId = UUID.randomUUID();
     snapshot =
         new Snapshot()
             .id(snapshotId)
-            .snapshotSources(
-                List.of(new SnapshotSource().dataset(new Dataset().id(UUID.randomUUID()))));
+            .snapshotSources(List.of(new SnapshotSource().dataset(new Dataset().id(datasetId))));
     step =
         new DeleteSnapshotPopAndLockDatasetStep(
             snapshotId, snapshotService, datasetService, TEST_USER, sharedLock);
   }
 
+  // snapshot and dataset exist
   @Test
-  void setAuthDomainGroups() {
+  void doStep() {
     when(snapshotService.retrieve(snapshotId)).thenReturn(snapshot);
-    when(snapshotService.retrieveAuthDomains(eq(snapshotId), any()))
-        .thenReturn(List.of("group1", "group2"));
     var flightMap = new FlightMap();
     when(flightContext.getWorkingMap()).thenReturn(flightMap);
-    step.doStep(flightContext);
+    assertThat(
+        "Step is successful",
+        step.doStep(flightContext),
+        equalTo(StepResult.getStepResultSuccess()));
 
     FlightMap map = flightContext.getWorkingMap();
-    List<String> authDomains =
-        map.get(SnapshotWorkingMapKeys.SNAPSHOT_AUTH_DOMAIN_GROUPS, new TypeReference<>() {});
+    assertEquals(true, map.get(SnapshotWorkingMapKeys.SNAPSHOT_EXISTS, Boolean.class));
```

Comment:
We don't usually write `assertEquals(true` or `assertEquals(false`. You can use `assertTrue()` or `assertFalse()` instead here
```suggestion
    assertTrue(map.get(SnapshotWorkingMapKeys.SNAPSHOT_EXISTS, Boolean.class));
```

Reply:
I did that originally, but I got this warning: Unboxing of 'map.get(SnapshotWorkingMapKeys.SNAPSHOT_EXISTS, Boolean.class)' may produce 'NullPointerException'. So, thats why I ended up doing this, but I agree its not ideal.

Reply:
Okay I used assertTrue/assertFalse and requireNonNullOrElse and set the else to the boolean I was not expecting. what do you think of this approach?

Reply:
Ugh, I forgot that sonar is so strict about boxed types. All things being equal we're probably better off with your original code. Or, you could define a helper method
```
  static boolean getBoolean(FlightMap map, String key) {
    return Objects.requireNonNull(map.get(key, Boolean.class));
  }

...
    assertTrue(getBoolean(map, SnapshotWorkingMapKeys.SNAPSHOT_EXISTS));
    assertTrue(getBoolean(map, SnapshotWorkingMapKeys.SNAPSHOT_HAS_GOOGLE_PROJECT));
...
```